src/screens/new-wallet/ImportWalletSeedScreen.tsx

```diff
@@ -17,43 +17,76 @@ along with the library. If not, see <http://www.gnu.org/licenses/>.
 */
 
 import { StackScreenProps } from '@react-navigation/stack'
-import { useEffect, useState } from 'react'
-import styled from 'styled-components/native'
+import { colord } from 'colord'
+import { useEffect, useRef, useState } from 'react'
+import { Pressable, ScrollView } from 'react-native'
+import Animated, { FadeIn, FadeOut, Layout } from 'react-native-reanimated'
+import styled, { useTheme } from 'styled-components/native'
 
+import AppText from '../../components/AppText'
 import Button from '../../components/buttons/Button'
 import Input from '../../components/inputs/Input'
-import Screen from '../../components/layout/Screen'
-import CenteredInstructions, { Instruction } from '../../components/text/CenteredInstructions'
+import Screen, { ScreenSection, ScreenSectionTitle } from '../../components/layout/Screen'
 import { useAppDispatch, useAppSelector } from '../../hooks/redux'
 import useBiometrics from '../../hooks/useBiometrics'
 import useOnNewWalletSuccess from '../../hooks/useOnNewWalletSuccess'
 import RootStackParamList from '../../navigation/rootStackRoutes'
 import { walletStored } from '../../store/activeWalletSlice'
 import { importedMnemonicChanged } from '../../store/walletGenerationSlice'
+import { BORDER_RADIUS, BORDER_RADIUS_SMALL } from '../../style/globalStyle'
+import { bip39Words } from '../../utils/bip39'
 
 type ScreenProps = StackScreenProps<RootStackParamList, 'NewWalletNameScreen'>
 
+type SelectedWord = {
+  word: string
+  timestamp: Date
+}
+
 const ImportWalletSeedScreen = ({ navigation }: ScreenProps) => {
   const dispatch = useAppDispatch()
-  const [secretPhrase, setSecretPhrase] = useState('')
-  const [words, setWords] = useState<string[]>([])
   const walletName = useAppSelector((state) => state.walletGeneration.walletName)
   const activeWallet = useAppSelector((state) => state.activeWallet)
   const hasAvailableBiometrics = useBiometrics()
+  const [typedInput, setTypedInput] = useState('')
+  const [selectedWords, setSelectedWords] = useState<SelectedWord[]>([])
+  const [possibleMatches, setPossibleMatches] = useState<string[]>([])
+  const theme = useTheme()
+  const scrollRef = useRef<ScrollView>(null)
+  const allowedWords = useRef(bip39Words.split(' '))
 
   useEffect(() => {
-    setWords(
-      secretPhrase
-        .trim()
-        .split(' ')
-        .filter((word) => word.length > 2)
+    setPossibleMatches(
+      typedInput.length <= 2
+        ? []
+        : allowedWords.current.filter((allowedWord) => allowedWord.startsWith(typedInput.toLowerCase().trim()))
+    )
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
I've seen you doing this twice, which in my mind is not a good practice :) Have you read more on the subject?
Is it better to ignore the warning or to compare the previous and current state before executing this effect?

Reply:
This is leftovers from my previous code, I should have removed the line!